### DISCUSSION TITLE
Welcome to StaticCodeEvalTools Discussions!

### DISCUSSION THREAD

**Discussion by pacmaninbw** [pacmaninbw/StaticCodeEvalTools#10] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by pacmaninbw** [comment]
I'm considering moving all the program dependent code from CommandLineParser to ProgramOptions. Your thoughts are appreciated!

**Reply by tms13** [reply]
I'm actually working on a command-line argument handler, somewhat inspired by Python's `argparse` module.  I'm hoping it will be suitable for inclusion into this program when it's done.  I intend to put it on Code Review this week.

**Reply by pacmaninbw** [reply]
I did make the changes I was considering but it was in a branch that I would like you and Edward to look at. I did this before you replied. I'm looking forward to reviewing the your solution and possibly including it in this project.

**Reply by beroset** [reply]
I did a cursory look and I believe it's an improvement.  I like the suggestion of modeling after `argparse`.  It's a tidy interface.

**Comment by pacmaninbw** [comment]
Toby, I merged my changes in. One of the changes is that the files will be in the correct order now. So if you do 'wc *.h *.cpp' all the .h files in the directory will be processed before all the .cpp files. I used std::ranges and std::views to implement this. I believe there is a 30% performance hit for this change. Command line parsing for 565 files went from just over 1 second to 1.4 seconds. It iterates over the directory twice to find the correct files.

**Reply by pacmaninbw** [reply]
You can try testing it.

**Comment by pacmaninbw** [comment]
I have opened 5 issues based on testing or issue found in the code review.

**Comment by pacmaninbw** [comment]
What is interesting is that neither of you are suggesting the boost library program options.

**Reply by tms13** [reply]
That's a good idea.  I've never looked at it myself.  And if you're bringing in Boost, you could consider its mmap() abstraction, as I alluded to on CR.

**Reply by pacmaninbw** [reply]
I looked at it before I wrote my command line parser. I couldn't figure out how to use the program options in the rest of the program.

**Comment by pacmaninbw** [comment]
I updated the CMakelists.txt file so that it works on Windows. Let me know if I broke the Linux side of things.

**Comment by pacmaninbw** [comment]
Any recommendations for open source test coverage tools?
